File: files/CH-224.py
import pandas as pd

path = "CH-224 Column Splitting.xlsx"

input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=6)

result = input["ID"].str.split("-", expand=True)
result.columns = ["ID1", "ID2", "ID3", "ID4", "ID5"]

result["ID.1"] = result[["ID1", "ID2"]].apply(lambda x: "-".join(x.dropna()), axis=1)
result["ID.2"] = result[["ID3", "ID4", "ID5"]].apply(lambda x: "-".join(x.dropna()), axis=1)

result = result[["ID.1", "ID.2"]]

print(result.equals(test)) # True

File: files/200-299/224/CH-224.R
library(tidyverse)
library(readxl)

path = "files/CH-224 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B7")
test = read_excel(path, range = "D2:E7")

result = input %>%
  separate(
    col = "ID",
    into = c("ID1", "ID2", "ID3", "ID4"),
    sep = "-",
    extra = "merge"
  ) %>%
  unite(col = "ID.1", ID1, ID2, sep = "-", na.rm = TRUE) %>%
  unite(col = "ID.2", ID3, ID4, sep = "-", na.rm = TRUE)

all.equal(result, test)
# [1] TRUE
